Web/Scotty/Trans/Lazy.hs

```diff
@@ -37,6 +38,18 @@ raiseStatus s = Base.raiseStatus s . T.toStrict
 redirect :: (Monad m) => T.Text -> ActionT m a
 redirect = Base.redirect . T.toStrict
 
+-- | Redirect to given URL with the specified status.
+-- Like throwing an uncatchable exception. Any code
+-- after the call to redirect will not be run.
+--
+-- > redirectStatus status303 "http://www.google.com"
+--
+-- OR
+--
+-- > redirectStatus status303 "/foo/bar"
+redirectStatus :: (Monad m) => Status -> T.Text -> ActionT m a
```

Comment:
This is not constrained enough: one could redirect with a 200 or whatever, which would not make much sense.

Reply:
I think that running a check here and throwing the exception would create a bad user experience. It might be nicer to have a specific ‘redirectX’ functions for each status code, with the bare ‘redirect’ remaining as the 302 version. But that would require quite a few new functions.

@ocramz what are your thoughts on a path forward?

Reply:
@hackeryarn The error codes for redirect are not even that many: 300-307, 307, 308 https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages 

I think overall providing them as individual functions would make for more solid API design (e.g `redirect301`, `redirect302` etc.). WDYT?

Reply:
That’s the direction I was leaning to, just wanted to make sure it wouldn’t clutter the API too much.

I’ll get this updated with the new functions.

Reply:
@ocramz This should do it. I skipped codes `305` and `306` since they are both deprecated. I also moved all the redirects to their own docs section, that way it's easier for users to find everything they are looking for.